woocommerce/woocommerce-ios
# [ExPlat] Add A/A test in the logged out state

Closes: #7435
Experiment design ref: pbxNRc-1S0-p2



### Description


In order to run A/B tests on future experiments to increase the login rate for WooActivate, we need to run an A/A test first to make sure there is no bias in the assignment. We pick a few metrics, launch an experiment without visual differences, and hope to see an even distribution of the two variants (`control` and `treatment`).

This PR removed the logged-in check in `AppDelegate` on app launch to start and refresh the A/B test setup, and also added a new A/A test enum case. In `AppDelegate`, `checkForUpgrades` is moved to after A/B testing setup because it fires the exposure event for the A/A test and needs to be called after the A/B test setup `ABTest.start()` (confirmed with Aaron@data in pbxNRc-1S0-p2#comment-3631).

There is likely a merge conflict with another A/A test https://github.com/woocommerce/woocommerce-ios/pull/7411, and I'll fix any merge conflicts before merging.

### Testing instructions


It's not trivial to test the logged-out state because of the random anonymous ID, @jaclync will make sure to check the following from [Aaron's CR feedback on the logged-in A/A test PR](https://github.com/woocommerce/woocommerce-ios/pull/7411#issuecomment-1211045219):

- [x] Make sure users can be assigned both `control` and `treatment` using the Abacus bookmarklet PCYsg-Fq7-p2#abacus-assignment-bookmarklet to manually assign yourself to each `treatment` variation
  - Note: I had to reverse-engineer the JS script and make a PATCH API request locally with sandboxed API
- [x] For the exposure event, make sure it fires after the assignment. If not, then we can either remove the exposure event from the experiment model in Abacus or choose a different exposure event. Since this is an A/A experiment, it doesn't matter too much to have or not have an exposure event

---
